tests/generation/test_utils.py

```diff
@@ -2285,6 +2285,12 @@ def _test_attention_implementation(self, attn_implementation):
                     inputs_dict[input_name] = input_data
             main_input = inputs_dict[model_class.main_input_name]
 
+            # FA2 doesn't accept masking in the middle of the sequence for now
+            if attn_implementation == "flash_attention_2":
+                for input_name in ("attention_mask", "decoder_attention_mask", "encoder_attention_mask"):
+                    if input_name in inputs_dict:
+                        inputs_dict[input_name] = torch.ones_like(inputs_dict[input_name])
```

Comment:
I am not able to see easily the relation of this change and the comment

> generate + FA2 test -> don't pass attention masks with right-padding (they are only equivalent with left-padding)

if you can explain it a bit more for me 🙏 .

Reply:
FA2 doesn't support all types of attention masks, and the support depends on the version of FA2. See `flash_attn_supports_top_left_mask` 👀 

The `attention_mask` in the tests is often right-padded, e.g. `input_mask = torch.tril(torch.ones_like(input_ids).to(torch_device))`

Reply:
(also added comment to test)